airflow/serialization/serialized_objects.py

```diff
@@ -424,7 +424,10 @@ def serialize_operator(cls, op: BaseOperator) -> Dict[str, Any]:
                     )
 
                 deps.append(f'{module_name}.{klass.__name__}')
-            serialize_op['deps'] = deps
+            # deps needs to be sorted here, because op.deps is a set, which is unstable when traversing,
+            # and the same call may get different results.
+            # When calling json.dumps(self.data, sort_keys=True) to generate dag_hash, misjudgment will occur
+            serialize_op['deps'] = sorted(deps)
```

Comment:
Are you using Custom deps? Can you please update the PR description and title to better reflect the changes now, something along the lines of, Handle Custom deps correct in DAG Serialization or something like that

Reply:
Thank you for your comments, I have updated the PR description and title.